sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -30,21 +30,44 @@ import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 
 /**
+ * Common base class for time zone aware expressions.
+ */
+trait TimeZoneAwareExpression extends Expression {
+
+  def zoneId: String
+
+  def timeZoneResolved: Boolean = zoneId != null
+
+  def withTimeZone(zoneId: String): TimeZoneAwareExpression
+
+  def timeZone: TimeZone = TimeZone.getTimeZone(zoneId)
```

Comment:
should this be a lazy val? otherwise it is pretty expensive to keep creating a new timezone object per row in the interpreted path.